akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/users/MoveUserToGuestLobbyCmdMsgHdlr.scala

```diff
@@ -0,0 +1,82 @@
+package org.bigbluebutton.core.apps.users
+
+import org.bigbluebutton.common2.msgs._
+import org.bigbluebutton.core.models._
+import org.bigbluebutton.core.running.{ LiveMeeting, OutMsgRouter }
+import org.bigbluebutton.core2.message.senders.{ Sender, MsgBuilder }
+import org.bigbluebutton.core.apps.{ PermissionCheck, RightsManagementTrait }
+
+trait MoveUserToGuestLobbyCmdMsgHdlr extends RightsManagementTrait {
+  this: UsersApp =>
+
+  val liveMeeting: LiveMeeting
+  val outGW: OutMsgRouter
+
+  def handleMoveUserToGuestLobbyCmdMsg(msg: MoveUserToGuestLobbyCmdMsg) {
+    val meetingId = liveMeeting.props.meetingProp.intId
+
+    if (permissionFailed(
+      PermissionCheck.MOD_LEVEL,
+      PermissionCheck.VIEWER_LEVEL,
+      liveMeeting.users2x,
+      msg.header.userId
+    ) || liveMeeting.props.meetingProp.isBreakout) {
+      val reason = "No permission to move user to guest lobby."
+      PermissionCheck.ejectUserForFailedPermission(meetingId, msg.header.userId, reason, outGW, liveMeeting)
+    } else {
+
+      // remove from Users list
+      for {
+        user <- Users2x.ejectFromMeeting(liveMeeting.users2x, msg.body.userMovedToGuestLobbyId)
+      } yield {
+
+        //assign presenter
+        if (user.presenter) {
+          UsersApp.automaticallyAssignPresenter(outGW, liveMeeting)
+        }
+        UsersApp.sendEjectUserFromVoiceToFreeswitch(
+          outGW,
+          liveMeeting.props.meetingProp.intId,
+          liveMeeting.props.voiceProp.voiceConf, user.intId
+        )
+      }
+
+      // Eject user from the conference
+      for {
+        userMoved <- RegisteredUsers.findWithUserId(msg.body.userMovedToGuestLobbyId, liveMeeting.registeredUsers)
+        movedByUser <- RegisteredUsers.findWithUserId(msg.body.movedToGuestLobbyBy, liveMeeting.registeredUsers)
+      } yield {
+        RegisteredUsers.setWaitingForApproval(liveMeeting.registeredUsers, userMoved, GuestStatus.WAIT)
+
+        val reason = "Moved to guest lobby";
+        UsersApp.sendUserEjectedMessageToClient(outGW, meetingId, userMoved.id, movedByUser.id, reason, EjectReasonCode.MOVED_TO_GUEST_LOBBY);
+        UsersApp.sendUserLeftMeetingToAllClients(outGW, meetingId, userMoved.id)
+        UsersApp.sendEjectUserFromSfuSysMsg(outGW, meetingId, userMoved.id)
+        Sender.sendDisconnectClientSysMsg(meetingId, userMoved.id, movedByUser.id, EjectReasonCode.MOVED_TO_GUEST_LOBBY, outGW)
+      }
+
+      //Notify Moderator about new guest and register the user anew
+      for {
+        movedUser <- RegisteredUsers.findWithUserId(msg.body.userMovedToGuestLobbyId, liveMeeting.registeredUsers)
+      } yield {
+        val movedUserToRegister = MovedUserToGuestLobby(movedUser.id, movedUser.externId, movedUser.name,
+          movedUser.role, movedUser.authToken, movedUser.avatarURL,
+          true, movedUser.authed, GuestStatus.WAIT,
```

Comment:
```suggestion
          movedUser.guest, movedUser.authed, GuestStatus.WAIT,
```

Reply:
I believe this should be reverted back to use the registered user guest property.

Basically BBB has 2 ways of thinking on who's a guest in a session. When `authenticatedGuest` is set true at bbb-web (the default), any user that is not a moderator is considered a "guest" and if the user joins as `authenticatedGuest` he is treated as an unauthenticated user. https://github.com/bigbluebutton/bigbluebutton/blob/8404048cfd35280d2dd427fc0c2e93790760c3f7/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy#L214-L223

On the other hand, when `authenticatedGuest` is set to false, BigBlueButton "guests" will only be the ones that join as `guest=true`. Personally I find this approach much more easy to understand and I really hope BBB will, someday, ship this as the default.

Reply:
Following this overview, I believe the solution for the `authenticatedGuest=false` setup must be on checking it at the client side.